File: ParkHaeYoung/02월 1-2주차/시소짝꿍.js
function solution(weights) {
    let answer = 0;

    const store = {};
    const ratio = [1, 3/2, 2, 4/3];

    weights.sort((a, b) => b - a).forEach((weight) => {
        ratio.forEach((el) => {
            const key = weight * el;
            if(store[key]) {
                answer += store[key];
            }
        });
        if(!store[weight]) store[weight] = 1;
        else store[weight]++;
    })

    return answer;
}

File: LeeNaYoung/2월 1주차/시소짝꿍.py
def solution(weights):
    from collections import defaultdict # 사용했을 때 없으면 int-> 0으로 세팅
    answer = 0
    info = defaultdict(int)
    # typs = [[1,1],[2,1],[3,2],[4,3]]
    # info로 불렀을 때 2번째 부터 1씩 추가
    for w in weights:
        answer += info[w] + info[w*2] + info[w/2] + info[(w*2)/3] + info[(w*3)/2] + info[(w*4)/3] + info[(w*3)/4]
        info[w] += 1
    return answer
